### DISCUSSION TITLE
Please help me to understand bucket system

### DISCUSSION THREAD

**Discussion by ykurilov** [kohya-ss/sd-scripts#632] [Q&A] [answered]
I've started my training. I use 1024 sized photos and choose 768 resolution. I've expected that different buckets will be created, but looks like they didn't. Because in log it says only about bucket 0.. 

So I want to understand how does it work. Thank you!

![image](https://github.com/kohya-ss/sd-scripts/assets/110710320/2c9d4768-b336-4e1c-a95c-90d0d0b72892)

**Comment by TingTingin** [comment, marked as answer]
The buckets are based on image aspect ratio if the aspect ratio is the same or similar they will be placed in the same bucket if the aspect ratios are different you will get different buckets

**Reply by ykurilov** [reply]
Thank you! So if all my images have the same aspect ratio, the will be put into one bucket? Ok. So what will happen to them if they are 1024x1024 and the training resolution is 512? It will just downscale them all to 512x512? 

And if they are 1024x768? Then what happen?

**Reply by TingTingin** [reply]
Yes

**Reply by ykurilov** [reply]
So what will happen to 1024x768 image if I train with resolution 512?

**Reply by TingTingin** [reply]
I can't remember the exasxt math but will most likely scale to a size like 576x448 etc
